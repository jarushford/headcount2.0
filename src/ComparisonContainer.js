import React from 'react'
import PropTypes from 'prop-types'
import District from './District'
import ComparisonWindow from './ComparisonWindow'
import './styles/ComparisonContainer.css'

export default function ComparisonContainer({
  compared1, compared2, comparedAvg, compareDistrict, currentData
}) {
  const comparisonPlaceholder = (
    <div className="comparison-placeholder">
      Select a district to compare
    </div>
  )

  let compared1Component
  let compared2Component

  if (compared1) {
    compared1Component = (
      <District
        districtData={compared1}
        compareDistrict={compareDistrict}
        selected
      />
    )
  } else {
    compared1Component = comparisonPlaceholder
  }

  if (compared2) {
    compared2Component = (
      <District
        districtData={compared2}
        compareDistrict={compareDistrict}
        selected
      />
    )
  } else {
    compared2Component = comparisonPlaceholder
  }

  return (
    <section className="comparison-container">
      {compared1Component}
      <ComparisonWindow
        compared1={compared1}
        compared2={compared2}
        comparedAvg={comparedAvg}
        currentData={currentData}
      />
      {compared2Component}
    </section>
  )
}

ComparisonContainer.propTypes = {
  compared1: PropTypes.shape({ location: '', stats: PropTypes.shape({ 2006: 0 }) }),
  compared2: PropTypes.shape({ location: '', stats: PropTypes.shape({ 2006: 0 }) }),
  comparedAvg: PropTypes.shape({ Location: 0, Location2: 0, compared: 0 }),
  currentData: PropTypes.string.isRequired,
  compareDistrict: PropTypes.func.isRequired
}

ComparisonContainer.defaultProps = {
  compared1: undefined,
  compared2: undefined,
  comparedAvg: undefined
}
