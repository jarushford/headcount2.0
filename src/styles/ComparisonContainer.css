.comparison-container {
  margin: 40px;
  display: flex;
  border-bottom: 4px solid rgb(46, 46, 46);
  padding-bottom: 40px;
}

.comparison-placeholder {
  border-radius: 3px;
  border: 2px solid #333;
  box-shadow: 1px 1px 5px 1px rgba(0, 0, 0, 0.3);
  padding: 4px;
  width: 315px;
  margin: 0 auto;
  height: 440px;
  display: flex;
  align-items: center;
  font-size: 1.3rem;
  font-family: 'Chalk';
  color: white;
  text-align: center;
  padding: 15px;
}

.comparison-window {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 20%;
  height: 400px;
  background: rgb(46, 46, 46);
  border-radius: 3px;
  box-shadow: 1px 1px 5px 1px rgba(0, 0, 0, 0.3);
  color: white;
  padding: 15px;
  margin-top: 20px;
}

.none-selected {
  background: none;
  box-shadow: none;
}

.comparison-window p,
.comparison-window h3 {
  text-align: center;
  font-size: 1.3rem;
  font-family: monospace;
}

.comparison-window h3 {
  font-size: 1.5rem;
  font-family: 'Chalk';
  letter-spacing: 2px;
}

.comparison-window h2 {
  text-align: center;
  margin-left: 0;
}