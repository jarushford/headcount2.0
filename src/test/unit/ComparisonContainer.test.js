import React from 'react';
import ComparisonContainer from '../../ComparisonContainer';
import { shallow } from 'enzyme'

describe('Comparison Container', () => {
  const compared1 = {
    location: 'Colorado',
    stats: { 2006: 0.5 }, 
    districtAvg: 0.5 
  }
  const compared2 = {
    location: 'ASPEN', 
    stats: { 2006: 0.8 }, 
    districtAvg: 0.8 
  }
  const avg = {
    'Colorado': 0.5, 
    'ASPEN': 0.8, 
    comparedd: 0.65
  }
  const mockcompare = jest.fn()

  it('should match the snapshot with or without data', () => {
    const wrapper1 = shallow(<ComparisonContainer compared1={compared1} compared2={compared2} comparedAvg={avg} comparedDistrict={mockcompare}/>)
    const wrapper2 = shallow(<ComparisonContainer comparedDistrict={mockcompare}/>)
    const wrapper3 = shallow(<ComparisonContainer compared1={compared1} compared2={undefined} comparedAvg={avg} comparedDistrict={mockcompare}/>)
    const wrapper4 = shallow(<ComparisonContainer compared1={undefined} compared2={compared2} comparedAvg={avg} compareDistrict={mockcompare}/>)

    expect(wrapper1).toMatchSnapshot()
    expect(wrapper2).toMatchSnapshot()
    expect(wrapper3).toMatchSnapshot()
    expect(wrapper4).toMatchSnapshot()
  })

})