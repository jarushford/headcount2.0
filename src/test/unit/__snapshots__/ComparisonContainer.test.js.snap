// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparison Container should match the snapshot with or without data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    compared1={
        Object {
            "districtAvg": 0.5,
            "location": "Colorado",
            "stats": Object {
              "2006": 0.5,
            },
          }
    }
    compared2={
        Object {
            "districtAvg": 0.8,
            "location": "ASPEN",
            "stats": Object {
              "2006": 0.8,
            },
          }
    }
    comparedAvg={
        Object {
            "ASPEN": 0.8,
            "Colorado": 0.5,
            "comparedd": 0.65,
          }
    }
    comparedDistrict={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <District
          compareDistrict={undefined}
          districtData={
                    Object {
                              "districtAvg": 0.5,
                              "location": "Colorado",
                              "stats": Object {
                                "2006": 0.5,
                              },
                            }
          }
          selected={true}
/>,
        <ComparisonWindow
          compared1={
                    Object {
                              "districtAvg": 0.5,
                              "location": "Colorado",
                              "stats": Object {
                                "2006": 0.5,
                              },
                            }
          }
          compared2={
                    Object {
                              "districtAvg": 0.8,
                              "location": "ASPEN",
                              "stats": Object {
                                "2006": 0.8,
                              },
                            }
          }
          comparedAvg={
                    Object {
                              "ASPEN": 0.8,
                              "Colorado": 0.5,
                              "comparedd": 0.65,
                            }
          }
/>,
        <District
          compareDistrict={undefined}
          districtData={
                    Object {
                              "districtAvg": 0.8,
                              "location": "ASPEN",
                              "stats": Object {
                                "2006": 0.8,
                              },
                            }
          }
          selected={true}
/>,
      ],
      "className": "comparison-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "compareDistrict": undefined,
          "districtData": Object {
            "districtAvg": 0.5,
            "location": "Colorado",
            "stats": Object {
              "2006": 0.5,
            },
          },
          "selected": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "compared1": Object {
            "districtAvg": 0.5,
            "location": "Colorado",
            "stats": Object {
              "2006": 0.5,
            },
          },
          "compared2": Object {
            "districtAvg": 0.8,
            "location": "ASPEN",
            "stats": Object {
              "2006": 0.8,
            },
          },
          "comparedAvg": Object {
            "ASPEN": 0.8,
            "Colorado": 0.5,
            "comparedd": 0.65,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "compareDistrict": undefined,
          "districtData": Object {
            "districtAvg": 0.8,
            "location": "ASPEN",
            "stats": Object {
              "2006": 0.8,
            },
          },
          "selected": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <District
            compareDistrict={undefined}
            districtData={
                        Object {
                                    "districtAvg": 0.5,
                                    "location": "Colorado",
                                    "stats": Object {
                                      "2006": 0.5,
                                    },
                                  }
            }
            selected={true}
/>,
          <ComparisonWindow
            compared1={
                        Object {
                                    "districtAvg": 0.5,
                                    "location": "Colorado",
                                    "stats": Object {
                                      "2006": 0.5,
                                    },
                                  }
            }
            compared2={
                        Object {
                                    "districtAvg": 0.8,
                                    "location": "ASPEN",
                                    "stats": Object {
                                      "2006": 0.8,
                                    },
                                  }
            }
            comparedAvg={
                        Object {
                                    "ASPEN": 0.8,
                                    "Colorado": 0.5,
                                    "comparedd": 0.65,
                                  }
            }
/>,
          <District
            compareDistrict={undefined}
            districtData={
                        Object {
                                    "districtAvg": 0.8,
                                    "location": "ASPEN",
                                    "stats": Object {
                                      "2006": 0.8,
                                    },
                                  }
            }
            selected={true}
/>,
        ],
        "className": "comparison-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "compareDistrict": undefined,
            "districtData": Object {
              "districtAvg": 0.5,
              "location": "Colorado",
              "stats": Object {
                "2006": 0.5,
              },
            },
            "selected": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "compared1": Object {
              "districtAvg": 0.5,
              "location": "Colorado",
              "stats": Object {
                "2006": 0.5,
              },
            },
            "compared2": Object {
              "districtAvg": 0.8,
              "location": "ASPEN",
              "stats": Object {
                "2006": 0.8,
              },
            },
            "comparedAvg": Object {
              "ASPEN": 0.8,
              "Colorado": 0.5,
              "comparedd": 0.65,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "compareDistrict": undefined,
            "districtData": Object {
              "districtAvg": 0.8,
              "location": "ASPEN",
              "stats": Object {
                "2006": 0.8,
              },
            },
            "selected": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Comparison Container should match the snapshot with or without data 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    comparedDistrict={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="comparison-placeholder"
>
          Select a district to compare
</div>,
        <ComparisonWindow
          compared1={
                    Object {
                              "districtAvg": "",
                              "location": "",
                              "stats": Object {},
                            }
          }
          compared2={
                    Object {
                              "districtAvg": "",
                              "location": "",
                              "stats": Object {},
                            }
          }
          comparedAvg={
                    Object {
                              "compared": "--",
                            }
          }
/>,
        <div
          className="comparison-placeholder"
>
          Select a district to compare
</div>,
      ],
      "className": "comparison-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select a district to compare",
          "className": "comparison-placeholder",
        },
        "ref": null,
        "rendered": "Select a district to compare",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "compared1": Object {
            "districtAvg": "",
            "location": "",
            "stats": Object {},
          },
          "compared2": Object {
            "districtAvg": "",
            "location": "",
            "stats": Object {},
          },
          "comparedAvg": Object {
            "compared": "--",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select a district to compare",
          "className": "comparison-placeholder",
        },
        "ref": null,
        "rendered": "Select a district to compare",
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="comparison-placeholder"
>
            Select a district to compare
</div>,
          <ComparisonWindow
            compared1={
                        Object {
                                    "districtAvg": "",
                                    "location": "",
                                    "stats": Object {},
                                  }
            }
            compared2={
                        Object {
                                    "districtAvg": "",
                                    "location": "",
                                    "stats": Object {},
                                  }
            }
            comparedAvg={
                        Object {
                                    "compared": "--",
                                  }
            }
/>,
          <div
            className="comparison-placeholder"
>
            Select a district to compare
</div>,
        ],
        "className": "comparison-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select a district to compare",
            "className": "comparison-placeholder",
          },
          "ref": null,
          "rendered": "Select a district to compare",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "compared1": Object {
              "districtAvg": "",
              "location": "",
              "stats": Object {},
            },
            "compared2": Object {
              "districtAvg": "",
              "location": "",
              "stats": Object {},
            },
            "comparedAvg": Object {
              "compared": "--",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select a district to compare",
            "className": "comparison-placeholder",
          },
          "ref": null,
          "rendered": "Select a district to compare",
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Comparison Container should match the snapshot with or without data 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    compared1={
        Object {
            "districtAvg": 0.5,
            "location": "Colorado",
            "stats": Object {
              "2006": 0.5,
            },
          }
    }
    compared2={undefined}
    comparedAvg={
        Object {
            "ASPEN": 0.8,
            "Colorado": 0.5,
            "comparedd": 0.65,
          }
    }
    comparedDistrict={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <District
          compareDistrict={undefined}
          districtData={
                    Object {
                              "districtAvg": 0.5,
                              "location": "Colorado",
                              "stats": Object {
                                "2006": 0.5,
                              },
                            }
          }
          selected={true}
/>,
        <ComparisonWindow
          compared1={
                    Object {
                              "districtAvg": 0.5,
                              "location": "Colorado",
                              "stats": Object {
                                "2006": 0.5,
                              },
                            }
          }
          compared2={
                    Object {
                              "districtAvg": "",
                              "location": "",
                              "stats": Object {},
                            }
          }
          comparedAvg={
                    Object {
                              "ASPEN": 0.8,
                              "Colorado": 0.5,
                              "comparedd": 0.65,
                            }
          }
/>,
        <div
          className="comparison-placeholder"
>
          Select a district to compare
</div>,
      ],
      "className": "comparison-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "compareDistrict": undefined,
          "districtData": Object {
            "districtAvg": 0.5,
            "location": "Colorado",
            "stats": Object {
              "2006": 0.5,
            },
          },
          "selected": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "compared1": Object {
            "districtAvg": 0.5,
            "location": "Colorado",
            "stats": Object {
              "2006": 0.5,
            },
          },
          "compared2": Object {
            "districtAvg": "",
            "location": "",
            "stats": Object {},
          },
          "comparedAvg": Object {
            "ASPEN": 0.8,
            "Colorado": 0.5,
            "comparedd": 0.65,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select a district to compare",
          "className": "comparison-placeholder",
        },
        "ref": null,
        "rendered": "Select a district to compare",
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <District
            compareDistrict={undefined}
            districtData={
                        Object {
                                    "districtAvg": 0.5,
                                    "location": "Colorado",
                                    "stats": Object {
                                      "2006": 0.5,
                                    },
                                  }
            }
            selected={true}
/>,
          <ComparisonWindow
            compared1={
                        Object {
                                    "districtAvg": 0.5,
                                    "location": "Colorado",
                                    "stats": Object {
                                      "2006": 0.5,
                                    },
                                  }
            }
            compared2={
                        Object {
                                    "districtAvg": "",
                                    "location": "",
                                    "stats": Object {},
                                  }
            }
            comparedAvg={
                        Object {
                                    "ASPEN": 0.8,
                                    "Colorado": 0.5,
                                    "comparedd": 0.65,
                                  }
            }
/>,
          <div
            className="comparison-placeholder"
>
            Select a district to compare
</div>,
        ],
        "className": "comparison-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "compareDistrict": undefined,
            "districtData": Object {
              "districtAvg": 0.5,
              "location": "Colorado",
              "stats": Object {
                "2006": 0.5,
              },
            },
            "selected": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "compared1": Object {
              "districtAvg": 0.5,
              "location": "Colorado",
              "stats": Object {
                "2006": 0.5,
              },
            },
            "compared2": Object {
              "districtAvg": "",
              "location": "",
              "stats": Object {},
            },
            "comparedAvg": Object {
              "ASPEN": 0.8,
              "Colorado": 0.5,
              "comparedd": 0.65,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select a district to compare",
            "className": "comparison-placeholder",
          },
          "ref": null,
          "rendered": "Select a district to compare",
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Comparison Container should match the snapshot with or without data 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    compareDistrict={[Function]}
    compared1={undefined}
    compared2={
        Object {
            "districtAvg": 0.8,
            "location": "ASPEN",
            "stats": Object {
              "2006": 0.8,
            },
          }
    }
    comparedAvg={
        Object {
            "ASPEN": 0.8,
            "Colorado": 0.5,
            "comparedd": 0.65,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="comparison-placeholder"
>
          Select a district to compare
</div>,
        <ComparisonWindow
          compared1={
                    Object {
                              "districtAvg": "",
                              "location": "",
                              "stats": Object {},
                            }
          }
          compared2={
                    Object {
                              "districtAvg": 0.8,
                              "location": "ASPEN",
                              "stats": Object {
                                "2006": 0.8,
                              },
                            }
          }
          comparedAvg={
                    Object {
                              "ASPEN": 0.8,
                              "Colorado": 0.5,
                              "comparedd": 0.65,
                            }
          }
/>,
        <District
          compareDistrict={[Function]}
          districtData={
                    Object {
                              "districtAvg": 0.8,
                              "location": "ASPEN",
                              "stats": Object {
                                "2006": 0.8,
                              },
                            }
          }
          selected={true}
/>,
      ],
      "className": "comparison-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select a district to compare",
          "className": "comparison-placeholder",
        },
        "ref": null,
        "rendered": "Select a district to compare",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "compared1": Object {
            "districtAvg": "",
            "location": "",
            "stats": Object {},
          },
          "compared2": Object {
            "districtAvg": 0.8,
            "location": "ASPEN",
            "stats": Object {
              "2006": 0.8,
            },
          },
          "comparedAvg": Object {
            "ASPEN": 0.8,
            "Colorado": 0.5,
            "comparedd": 0.65,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "compareDistrict": [Function],
          "districtData": Object {
            "districtAvg": 0.8,
            "location": "ASPEN",
            "stats": Object {
              "2006": 0.8,
            },
          },
          "selected": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="comparison-placeholder"
>
            Select a district to compare
</div>,
          <ComparisonWindow
            compared1={
                        Object {
                                    "districtAvg": "",
                                    "location": "",
                                    "stats": Object {},
                                  }
            }
            compared2={
                        Object {
                                    "districtAvg": 0.8,
                                    "location": "ASPEN",
                                    "stats": Object {
                                      "2006": 0.8,
                                    },
                                  }
            }
            comparedAvg={
                        Object {
                                    "ASPEN": 0.8,
                                    "Colorado": 0.5,
                                    "comparedd": 0.65,
                                  }
            }
/>,
          <District
            compareDistrict={[Function]}
            districtData={
                        Object {
                                    "districtAvg": 0.8,
                                    "location": "ASPEN",
                                    "stats": Object {
                                      "2006": 0.8,
                                    },
                                  }
            }
            selected={true}
/>,
        ],
        "className": "comparison-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select a district to compare",
            "className": "comparison-placeholder",
          },
          "ref": null,
          "rendered": "Select a district to compare",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "compared1": Object {
              "districtAvg": "",
              "location": "",
              "stats": Object {},
            },
            "compared2": Object {
              "districtAvg": 0.8,
              "location": "ASPEN",
              "stats": Object {
                "2006": 0.8,
              },
            },
            "comparedAvg": Object {
              "ASPEN": 0.8,
              "Colorado": 0.5,
              "comparedd": 0.65,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "compareDistrict": [Function],
            "districtData": Object {
              "districtAvg": 0.8,
              "location": "ASPEN",
              "stats": Object {
                "2006": 0.8,
              },
            },
            "selected": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
